### ISSUE TITLE
Improve API reference

### ISSUE THREAD

**Issue by dkafri** [quantumlib/Cirq#3667]
**Is your feature request related to a use case or problem? Please describe.**
I would like to look at the API for the Moment class. I spent a while sifting through the new [Cirq documentation](https://quantumai.google/reference/python/cirq/all_symbols) but I could not find it. I tried the search bar but it was not there. I tried to look at the code on github but a [simple search](https://github.com/quantumlib/Cirq/search?q=Moment) does not reveal where the Moment class is defined. My last resort was to use my IDE (PyCharm) to find its definition, but some users do not have this capability.

**Describe the solution you'd like**
Update the documentation page to make it easy to search for class definitions.

**[optional] Describe alternatives/workarounds you've considered**
See above.

**[optional] Additional context (e.g. screenshots)**

**What is the urgency from your perspective for this issue? Is it blocking important work?**
P1 - I need this no later than the next release (end of quarter)

**Comment by rmlarose** [comment]
[`cirq.Moment`](https://quantumai.google/reference/python/cirq/ops/Moment) appears in the [API reference](https://quantumai.google/reference/python/cirq/all_symbols) for me:

![image](https://user-images.githubusercontent.com/32416820/104445806-971a2e80-5567-11eb-95ea-86f01d7f966b.png)

as well as by searching "Moment" on [quantumai.google](https://quantumai.google/):

![image](https://user-images.githubusercontent.com/32416820/104446021-d8aad980-5567-11eb-8ed6-9da26e037d92.png)

**Comment by maffoo** [comment]
It is a bit unfortunate that the "all symbols" page shows all of `cirq.Moment`, `cirq.ops.Moment`, and `cirq.ops.moment.Moment`. All of those link to a page showing `cirq.ops.Moment` with the other two (`cirq.Moment` and `cirq.ops.moment.Moment`) listed as aliases, even though `cirq.Moment` is the preferred way for users to access it.

**Comment by dkafri** [comment]
I guess I didn't think to scroll through that whole list at the front page. When I searched "Moment" in the search bar I did not see the link to the API reference.
![screenshot-2021-01-13-09-15-04](https://user-images.githubusercontent.com/30353657/104486039-10be1680-5580-11eb-808d-c1266ddd3e7a.png)

**Comment by 95-martin-orion** [comment]
Seems like the issue here is the discrepancy between the instant drop-down list and the search results, as I can reproduce both experiences described above. Oddly, the search seems to return just about every page _except_ the Moment page - even other files under `.../reference/python/cirq`. Is the `ops` directory somehow hidden from the search results?

**Comment by rmlarose** [comment]
Indeed there's a discrepancy between suggested searches and search results. I don't think `ops` is hidden however as it shows up with a [search for ops](https://quantumai.google/s/results?q=ops), albeit below several entries for the openfermion ops module.

![image](https://user-images.githubusercontent.com/32416820/105248926-41d0b500-5b45-11eb-932c-0fad6f483451.png)

**Comment by balopat** [comment]
Hi All, unfortunately we don't have too much control over this experience :(. Our users will have to get used to the dropdown behavior. Hitting enter takes you to a Google search of the site that is hard to influence, as it's based on the indexing of Search. On top of that this is not something that we can influence from this repo at all, but the internal devsite people might have some wisdom around it. I can share the internal bugtracking component with you, @dkafri offline. As we can't really do anything about it here, I would close this issue.

**Comment by balopat** [comment]
Actually there is one thing that we _can_ do: link to the mentions of the Moment class to the reference. Which is as simple as rewriting `Moment` to `cirq.Moment` everywhere. This will automatically get converted to a link pointing to the API docs. That way at least, if the Google search takes you to a page, you can still click into the link.

**Comment by 95-martin-orion** [comment]
> Actually there is one thing that we _can_ do: link to the mentions of the Moment class to the reference. Which is as simple as rewriting `Moment` to `cirq.Moment` everywhere.

Of course, since this presumably affects _all_ objects (not just Moment), the required change is much larger than this. Still simple, but likely to touch a lot more of the Cirq codebase.

**Comment by unaiic** [comment]
@balopat @95-martin-orion Do you mean rewriting all mentions to objects in the docs to `cirq.ObjectName`?

**Comment by 95-martin-orion** [comment]
> @balopat @95-martin-orion Do you mean rewriting all mentions to objects in the docs to `cirq.ObjectName`?

That's roughly what we're suggesting, but if you'd like to take this on I recommend making an example PR with a single file before changing all docs. We'll need to balance this issue with overall readability of the docs.

Note that objects referenced this way must appear in the top-level `cirq/__init__.py` file to be auto-linked.

**Comment by unaiic** [comment]
@95-martin-orion Fine, I'll work on this. Thank you for the notes.

**Comment by 95-martin-orion** [comment]
Switched `won't-fix` for `accepted` since a solution is in progress.

**Comment by Strilanc** [comment]
> Our users will have to get used to the dropdown behavior.

It seems like a contradiction in terms for users to have to "get used to" the behavior of *documentation*.

**Comment by mhucka** [comment]
This is an old issue. Looking at the documentation site now, as near as we can figure out, the issue has been addressed at some point since the time this ticket was opened. Closing as done.
